sheets/utils.py

```diff
@@ -106,22 +120,49 @@ def parse_raw_data(cls, row_index, raw_row_data):
 
         Args:
             row_index (int): The row index according to the spreadsheet (not zero-based)
-            raw_row_data (list of str): The raw row data
+            raw_row_data (iterable of str): The raw row data
 
         Returns:
             CouponRequestRow: The parsed data row
+
+        Raises:
+            SheetRowParsingException: Raised if the row could not be parsed
         """
-        return cls(
-            row_index=row_index,
-            purchase_order_id=raw_row_data[0].strip(),
-            coupon_name=raw_row_data[1].strip(),
-            num_codes=int(raw_row_data[2]),
-            product_text_id=raw_row_data[3].strip(),
-            company_name=raw_row_data[4],
-            activation=parse_sheet_date_str(item_at_index_or_none(raw_row_data, 5)),
-            expiration=parse_sheet_date_str(item_at_index_or_none(raw_row_data, 6)),
-            date_processed=parse_sheet_date_str(item_at_index_or_none(raw_row_data, 7)),
-        )
+        try:
+            return cls(
+                row_index=row_index,
+                purchase_order_id=raw_row_data[cls.PURCHASE_ORDER_COL_INDEX].strip(),
```

Comment:
Should the other indices below also be variables?

Reply:
They're almost guaranteed not to change and those indices aren't referred to anywhere else in the codebase. I think adding settings would just add clutter